reo/views.py

```diff
@@ -590,6 +597,27 @@ def chp_defaults(request):
                 size_class = CHP.default_chp_size_class[prime_mover]
             prime_mover_defaults = {param: prime_mover_defaults_all[prime_mover][param][size_class]
                                     for param in prime_mover_defaults_all[prime_mover].keys()}
+            if year is not None:
+                year = int(year)
+                # TODO put in "prime_mover" instead of hard-coded "recip_engine" for path (after adding other prime_mover unavailability periods)
+                chp_unavailability_path = os.path.join('input_files', 'CHP', 'recip_engine_unavailability_periods.csv')
+                chp_unavailability_periods = pd.read_csv(chp_unavailability_path)
+                if calendar.isleap(year):  # Remove last day of the year if leap year
+                    end_date = "12/31/"+str(year)
+                else:
+                    end_date = "1/1/"+str(year+1)
+                dt_profile = pd.date_range(start='1/1/'+str(year), end=end_date, freq="1H", closed="left")
+                chp_unavailability_hourly = pd.Series(np.zeros(8760), index=dt_profile)
+                for i in range(len(chp_unavailability_periods)):
```

Comment:
Since this `for` loop is both in views.py and validators.py maybe we could define it in one of the files (or another) and import it where needed? Maybe in utilities.py or something specific to CHP. This would avoid repeated code (DRY or "Don't Repeat Yourself" is a common axiom for code development).

Reply:
I moved this loop/calculation into a new function in `utilities.py`, named it to be more of a general calculation for this purpose, and imported this function for `validators.py` and `views.py`, in [this commit](https://github.com/NREL/REopt_Lite_API/pull/153/commits/9c8a19f0ebdb7a99f9e99fc888568845067dc8cc).